hack/validate-canary.sh

```diff
@@ -152,17 +155,36 @@ function run() {
     clean-canary
     generate-canary-manifest
     deploy-canary
-    # Tail the logs of the reporter to block until it completes. The report only logs the result file.
-    res=$(kubectl logs -f oci-cloud-controller-manager-canary -c oci-cloud-controller-manager-canary-test-reporter)
-    # Display the results.
-    echo "${res}"
-    # Grep the log to return an error code.
-    error=$(echo "${res}" | grep 'end_time' | cut -d':' -f 1)
-    if [ -z ${error} ]; then
-        exit 1
-    else
-        exit 0
-    fi
+
```

Comment:
The validation test itself is a little more involved now. We wait for canary to run CANARY_RUNS times. If we have any failures we consider it FAILED. If we have all successes then PASSED.  On completion we destroy the canary pod.

We let the CI destroy long running tests. This should also destroy any running canaries.